docs/integrations/sources/codefresh.md

```diff
@@ -0,0 +1,51 @@
+# Codefresh
+This connector integrates Codefresh with Airbyte, enabling seamless data synchronization for CI/CD analytics and pipeline monitoring. It provides streams like agents, builds, audit, analytics etc.
+
+## Authentication
+First, create an account on [codefresh](https://codefresh.io/) if you don't  already have one. Then follow [this](https://g.codefresh.io/api/#section/Authentication/API-key) step to get the API key.
+
+## Configuration
+
+| Input | Type | Description | Default Value |
+|-------|------|-------------|---------------|
+| `type` | `string` | Type.  |  |
+| `api_key` | `string` | API Key.  |  |
+| `account_id` | `string` | Account Id.  |  |
+| `account_name` | `string` | Account Name.  |  |
+| `report_name` | `string` | Report Name.  |  |
+| `granularity` | `string` | Granularity.  |  |
+| `date_range` | `array` | Date Range.  |  |
+| `id` | `string` | Id.  |  |
+| `entity_type` | `string` | Entity Type.  |  |
+| `entity_id` | `string` | Entity Id.  |  |
+
+## Streams
+| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
+|-------------|-------------|------------|---------------------|----------------------|
+| accounts |  | No pagination | ✅ |  ❌  |
+| account_settings |  | No pagination | ✅ |  ❌  |
+| agents | id | No pagination | ✅ |  ❌  |
+| builds |  | DefaultPaginator | ✅ |  ❌  |
+| audit |  | DefaultPaginator | ✅ |  ❌  |
+| analytics_metadata |  | DefaultPaginator | ✅ |  ❌  |
+| analytics_reports |  | DefaultPaginator | ✅ |  ❌  |
+| execution_contexts | docs | DefaultPaginator | ✅ |  ❌  |
+| contexts |  | DefaultPaginator | ✅ |  ❌  |
+| views |  | DefaultPaginator | ✅ |  ❌  |
+| projects |  | DefaultPaginator | ✅ |  ❌  |
+| pipelines | metadata | DefaultPaginator | ✅ |  ❌  |
+| step_types |  | DefaultPaginator | ✅ |  ❌  |
+| environments |  | DefaultPaginator | ✅ |  ❌  |
+| helm_repos |  | No pagination | ✅ |  ❌  |
+| favorites |  | No pagination | ✅ |  ❌  |
```

Comment:
check again the missing primary keys as this can indicate you're not using the right record selector.

Reply:
ok I've read all the above comments and will update you soon

Reply:
@marcosmarxm I've added primary keys to most of them but couldnt find proper field to set as primary key for few streams like environment, help_repos and analytics_report